completion/services.py

```diff
@@ -76,27 +76,31 @@ def fill_in_run(block_key):
                 completions[candidate] = 0.0
         return completions
 
-    def get_user_children(self, node):
+    def get_completable_children(self, node):
         """
         Verticals will sometimes have children that also have children (some
         examples are content libraries, split tests, conditionals, etc.). This
         function will recurse through the children to get to the bottom leaf
-        and return only those children.
+        and return only those children. This function heavily utilizes the
+        XBlockCompletionMode class to determine if it should recurse or not. It
+        will only recurse if the node is an AGGREGATOR
 
         Note: The nodes passed in should have already taken the user into
         account so the proper blocks are shown for this user.
         """
-        if node.scope_ids.block_type != 'discussion':
+        if XBlockCompletionMode.get_mode(node) == XBlockCompletionMode.EXCLUDED:
             return []
 
-        node_children = ((hasattr(node, 'get_child_descriptors') and node.get_child_descriptors())
-                         or (hasattr(node, 'get_children') and node.get_children()))
         user_children = []
-        if node_children:
+        if XBlockCompletionMode.get_mode(node) == XBlockCompletionMode.AGGREGATOR:
+            node_children = ((hasattr(node, 'get_child_descriptors') and node.get_child_descriptors())
+                             or (hasattr(node, 'get_children') and node.get_children()))
             for child in node_children:
-                user_children.extend(self.get_user_children(child))
-        else:
+                user_children.extend(self.get_completable_children(child))
+
+        if XBlockCompletionMode.get_mode(node) == XBlockCompletionMode.COMPLETABLE:
```

Comment:
minor nit and I assume `get_mode` is fast, but would it make sense to save the value here and use `elif` instead of `if` in this method, to avoid calling `get_mode` thrice?